src/watertap_contrib/reflo/analysis/multiperiod/ltmed_vagmd_semibatch/MED_VAGMD_semibatch_class.py

```diff
@@ -0,0 +1,823 @@
+                    f"The valid range is {input_ranges[i][0]} - {input_ranges[i][1]}."
+                )
+
+        if md_module_type == "AS7C1.5L" and md_high_brine_salinity:
+            _logger.info(
+                f"For module AS7C1.5L, when the final brine salinity is larger than 175.3 g/L,"
+                f"the operational parameters will be fixed at nominal condition"
+            )
+
+        mp = MultiPeriodModel(
+            n_time_points=n_time_points,
+            process_model_func=build_med_md_flowsheet,
+            linking_variable_func=get_variable_pairs,
+            initialization_func=fix_dof_and_initialize,
+            unfix_dof_func=unfix_dof,
+            outlvl=logging.WARNING,
+        )
+
+        processing_options = {
+            "med_feed_salinity": med_feed_salinity,
+            "med_feed_temp": med_feed_temp,
+            "med_steam_temp": med_steam_temp,
+            "med_capacity": med_capacity,
+            "med_recovery_ratio": med_recovery_ratio,
+            "md_feed_flow_rate": md_feed_flow_rate,
+            "md_cond_inlet_temp": md_cond_inlet_temp,
+            "md_evap_inlet_temp": md_evap_inlet_temp,
+            "md_module_type": md_module_type,
+            "md_cooling_system_type": md_cooling_system_type,
+            "md_cooling_inlet_temp": md_cooling_inlet_temp,
+            "md_high_brine_salinity": md_high_brine_salinity,
+            "dt": dt,
+            "batch_volume": batch_volume,
+            "phase": "processing",
+        }
+
+        model_options = {t: processing_options for t in range(n_time_points)}
+
+        # create the multiperiod object
+        mp.build_multi_period_model(
+            model_data_kwargs=model_options,
+            flowsheet_options=processing_options,
+            initialization_options=None,
+            unfix_dof_options=None,
+        )
+
+        return mp
+
+    def add_refilling_phase(self):
+        """
+        This function adds the refilling phase of the batch operation
+        and initializes it
+        """
+        self.dt_refilling_phase = Var(
+            initialize=120,
+            bounds=(None, None),
+            units=pyunits.s,
+            doc="Time interval of the refilling phase",
+        )
+
+        # Get the last period for refilling phase calculations
+        first_blk = self.mp.get_active_process_blocks()[0]
+        last_blk = self.mp.get_active_process_blocks()[-1]
+
+        @self.Constraint(doc="Calculate time interval of the refilling phase")
+        def eq_dt_refilling(b):
+            return (
+                first_blk.fs.med.brine_props[0].flow_vol_phase["Liq"]
+                - first_blk.fs.vagmd.permeate_props[0].flow_vol_phase["Liq"]
+            ) * b.dt_refilling_phase == pyunits.convert(
+                self.config.model_input["batch_volume"] * pyunits.L,
+                to_units=pyunits.m**3,
+            ) - last_blk.fs.volume_in_tank
+
```

Comment:
Seeing the constraints help make a little more sense out of the equation you put in the documentation, so probably that equation should have more clarity to reflect the different time instances.

Reply:
Sure, the time instances have been added in the subscript of each variable.